controllers/new-page-pane.js

```diff
@@ -1,31 +1,192 @@
+}
+
+/**
+ * create a new page based on the provided ID
+ * note: if successful, will redirect to the new page.
+ * otherwise, will display an error message
+ * @param {string} id
+ * @returns {Promise}
+ */
+function createPageByType(id) {
+  return edit.createPage(id)
+    .then(function (url) {
+      location.href = url;
+    })
+    .catch(function () {
+      progress.done('error');
+      progress.open('error', 'Error creating new page', true);
+    });
+}
+
+module.exports = function () {
+  function constructor(el) {
+    // useful elements
+    this.input = dom.find(el, '.filtered-input');
+    this.list = dom.find(el, '.filtered-items'),
+    this.items = dom.findAll(this.list, '.filtered-item');
 
-  function createPageByType(id) {
-    return edit.createPage(id)
-      .then(function (url) {
-        location.href = url;
-      })
-      .catch(function () {
-        progress.done('error');
-        progress.open('error', 'Issue with opening page.', true);
-      });
+    // give the pane a moment to animate in, then focus the input
+    setTimeout(() => this.input.focus(), 100);
+
+    // set the height so when we filter it won't jump around
+    this.list.style.height = getComputedStyle(this.list).height;
   }
 
   constructor.prototype = {
     events: {
-      '.primary-action click': 'onNewPageClick'
+      '.filtered-input keydown': 'onInputKeydown',
+      '.filtered-input keyup': 'onInputKeyup',
+      '.filtered-item keydown': 'onItemKeydown',
+      '.filtered-item keyup': 'onItemKeyup',
+      '.filtered-item click': 'onItemClick'
+    },
+
+    onInputKeydown: function (e) {
+      var key = keyCode(e),
+        available = getAvailable(this.items);
+
+      // simulate active states when pressing enter
+      if (key === 'enter' && available.length === 1) {
+        available[0].classList.add('active');
+      }
+    },
+
+    onInputKeyup: function (e) {
+      var input = this.input,
+        key = keyCode(e),
+        available = getAvailable(this.items);
+
+      // if it's down, transfer focus
+      // if it's enter, try to add component
+      // if it's esc, exit the pane
+      // if it's anything else, try to filter the list
+      // note: tab will work natively
+
+      if (key === 'down') {
+        focusFirst(available); // focus on first available item in list
+      } else if (key === 'enter' && available.length === 1) {
+        available[0].classList.remove('active');
+        e.preventDefault();
+        createPageByType(available[0].getAttribute('data-item-name'));
+      } else if (key === 'enter') {
+        input.classList.add('kiln-shake');
+        setTimeout(() => input.classList.remove('kiln-shake'), 301); // length of the animation + 1
+      } else if (key === 'esc') {
+        pane.close();
+      } else {
+        filter(input.value, this.items);
+      }
+    },
+
+    onItemKeydown: function (e) {
+      // simulate active states when pressing enter
+      if (keyCode(e) === 'enter') {
```

Comment:
I think this would be better to compare to the value from keyCode https://www.npmjs.com/package/keycode#maps